tm2/pkg/crypto/crypto.go

```diff
@@ -85,12 +85,8 @@ func (addr *Address) UnmarshalAmino(b32str string) (err error) {
 	return nil
 }
 
-func (addr Address) Compare(other Address) int {
-	bz1 := make([]byte, len(addr))
-	bz2 := make([]byte, len(other))
-	copy(bz1, addr[:])
-	copy(bz2, other[:])
-	return bytes.Compare(bz1, bz2)
+func (addr *Address) Compare(other Address) int {
+	return bytes.Compare(addr[:], other[:])
```

Comment:
Why not use a value receiver instead of the pointer receiver?

Reply:
Okay, so really I'm wondering if it's really more efficient to use `*Address`, why isn't `other` also a pointer?

It is indeed more efficient, but the benchmark can be dropped further if you make them both pointers.

Reply:
So for the method argument, I figured that it would make the ergonomics much harder to use as it'd require a variable and pointer to be made for each argument so I left it at just the pointer receiver and then using bytes.Compare(a[:], b[:]). However I have no problem making the argument a pointer.